AudioDeviceExample/AudioDevices/ExampleAVAudioEngineDevice.m

```diff
@@ -510,13 +629,34 @@ static OSStatus ExampleAVAudioEngineDeviceRecordCallback(void *refCon,
                              numFrames,
                              audioBufferList);
 
-    int8_t *audioBuffer = (int8_t *)audioBufferList->mBuffers[0].mData;
-    UInt32 audioBufferSizeInBytes = audioBufferList->mBuffers[0].mDataByteSize;
+    AudioBufferList *mixedAudioBufferList = (AudioBufferList*)malloc(sizeof(AudioBufferList));
```

Comment:
This memory should be pre-allocated because the call site is a real-time audio thread (ExampleAVAudioEngineDeviceRecordCallback).